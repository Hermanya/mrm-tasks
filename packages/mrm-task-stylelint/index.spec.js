/* eslint-disable no-console */
'use strict';

jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));
jest.mock('mrm-core/src/npm', () => ({
	install: jest.fn(),
}));

const getConfigGetter = require('../../mrm').getConfigGetter;
const vol = require('memfs').vol;
const task = require('./index');

const json = o => JSON.stringify(o, null, '  ');

const packageJson = json({
	name: 'unicorn',
	scripts: {
		test: 'jest',
	},
});

afterEach(() => vol.reset());

it('should add stylelint', () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(getConfigGetter({}));

	expect(vol.toJSON()).toMatchSnapshot();
});

it.skip('should install custom preset', () => {});
it.skip('should install extra dependencies', () => {});
