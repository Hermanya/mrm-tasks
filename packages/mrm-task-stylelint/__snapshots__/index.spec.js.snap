// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add stylelint 1`] = `
Object {
  "/.stylelintrc": "{
  \\"extends\\": \\"stylelint-config-standard\\",
  \\"rules\\": {
    \\"selector-pseudo-class-no-unknown\\": [
      true,
      {
        \\"ignorePseudoClasses\\": [
          \\"global\\"
        ]
      }
    ]
  }
}",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"test\\": \\"jest\\",
    \\"lint:css\\": \\"stylelint '**/*.css'\\",
    \\"pretest\\": \\"npm run lint:css\\"
  }
}",
}
`;
